src/VCR/Storage/AbstractStorage.php

```diff
@@ -54,10 +54,12 @@
      * If the cassetteName contains PATH_SEPARATORs, subfolders of the
      * cassettePath are autocreated when not existing.
      *
-     * @param string $cassettePath Path to the cassette directory.
-     * @param string $cassetteName Path to the cassette file, relative to the path.
+     * @param string  $cassettePath   Path to the cassette directory.
+     * @param string  $cassetteName   Path to the cassette file, relative to the path.
+     * @param string  $defaultContent Default data for this cassette if its not existing
+     * @param boolean $writeable      Whether the file must be writeable
      */
-    public function __construct($cassettePath, $cassetteName, $defaultContent = '[]')
+    public function __construct($cassettePath, $cassetteName, $defaultContent = '[]', $writeable = true)
```

Comment:
Is there really a need to introduce another variable? I would have moved the writeable check down to the `storeRecording` where stuff gets written. The downside is that the check would have to be added in the `Json` and `Yaml` storage. Am I missing something?